File: day12/task-manager-app/backend/src/main/frontend/src/api/taskService.js
import axios from "axios";

const API_URL = "http://localhost:8080/api/tasks";

export const taskService = {
  getAllTasks: () => axios.get(API_URL),
  createTask: (task) => axios.post(API_URL, task),
  updateTask: (id, task) => axios.put(`${API_URL}/${id}`, task),
  deleteTask: (id) => axios.delete(`${API_URL}/${id}`),
};

File: day14/task-manager/src/main/java/com/workshop/task_manager/service/TaskService.java
package com.workshop.task_manager.service;

import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Service;

import com.workshop.task_manager.entity.Task;
import com.workshop.task_manager.repository.TaskRepository;

import java.util.List;
import java.util.Optional;

@Service
public class TaskService {
    @Autowired
    private TaskRepository taskRepository;

    public List<Task> getAllTasks() {
        return taskRepository.findAll();
    }

    public Optional<Task> getTaskById(Long id) {
        return taskRepository.findById(id);
    }

    public Task createTask(Task task) {
        return taskRepository.save(task);
    }

    public Task updateTask(Task task) {
        return taskRepository.save(task);
    }

    public void deleteTask(Long id) {
        taskRepository.deleteById(id);
    }

    public List<Task> getTasksByStatus(Boolean completed) {
        return taskRepository.findByCompletedOrderByCreatedAtDesc(completed);
    }
}